static/app/views/projectsDashboard/index.tsx

```diff
@@ -81,6 +82,64 @@ function addProjectsToTeams(teams: Team[], projects: Project[]): TeamWithProject
   }));
 }
 
+function getFilteredProjectsBasedOnTeams({
```

Comment:
Since there's a significant amount of code dedicated to filtering the projects, I extracted it into a new function to make it clearer and more manageable